src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -350,6 +373,70 @@ public static EquationSystem create(LfNetwork network, VariableSet variableSet,
 
+        network.addListener(new LfNetworkListener() {
+            @Override
+            public void onVoltageControlChange(LfBus bus, boolean oldVoltageControl, boolean newVoltageControl) {
+                if (newVoltageControl) { // switch PQ/PV
+                    Equation qEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_Q);
+                    qEq.setActive(false);
+
+                    LfBus controlledBus = bus.getControlledBus().orElse(null);
+                    if (controlledBus != null) {
+                        updateControlledBus(controlledBus, equationSystem, variableSet);
+                    } else {
+                        Equation vEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_V);
+                        vEq.setActive(true);
+                    }
+                } else { // switch PV/PQ
+                    Equation qEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_Q);
+                    qEq.setActive(true);
+
+                    LfBus controlledBus = bus.getControlledBus().orElse(null);
+                    if (controlledBus != null) {
+                        updateControlledBus(controlledBus, equationSystem, variableSet);
+                    } else {
+                        Equation vEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_V);
+                        vEq.setActive(false);
+                    }
+                }
+            }
+
+            @Override
+            public void onPhaseControlModeChange(DiscretePhaseControl phaseControl, DiscretePhaseControl.Mode oldMode, DiscretePhaseControl.Mode newMode) {
+                boolean on = newMode != DiscretePhaseControl.Mode.OFF;
+
+                // activate/de-activate phase control equation
+                equationSystem.createEquation(phaseControl.getControlled().getNum(), EquationType.BRANCH_P)
+                        .setActive(on);
+
+                // de-activate/activate constant phase equation
+                equationSystem.createEquation(phaseControl.getController().getNum(), EquationType.BRANCH_ALPHA1)
+                        .setActive(!on);
+            }
+
+            @Override
+            public void onVoltageControlModeChange(DiscreteVoltageControl voltageControl, DiscreteVoltageControl.Mode oldMode, DiscreteVoltageControl.Mode newMode) {
+                if (newMode == DiscreteVoltageControl.Mode.OFF) {
+                    LfBus bus = voltageControl.getControlled();
+
+                    // de-activate transformer voltage control equation
+                    Equation t = equationSystem.createEquation(bus.getNum(), EquationType.BUS_V);
+                    t.setActive(false);
+
+                    for (LfBranch controllerBranch : bus.getDiscreteVoltageControl().getControllers()) {
+                        // de-activate r1 variable
+                        Variable r1 = variableSet.getVariable(controllerBranch.getNum(), VariableType.BRANCH_RHO1);
```

Comment:
Here you manage the variable `BRANCH_RHO1`, so why `BRANCH_ALPHA1` has disappeared?

Reply:
I reverted the phase control refactoring. I will add it in another PR and will explain better why.